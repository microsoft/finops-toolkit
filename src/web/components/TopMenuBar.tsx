import { Text, makeStyles, Image, Divider } from '@fluentui/react-components';

const useStyles = makeStyles({
  fullWidthContainer: {
    width: '100vw',
    height: '40px',
    marginTop: '8px',
    padding: 0,
  },
  commandBar: {
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    width: '100%',
    backgroundColor: '#f4f6f8',
    overflowX: 'hidden',
    '@media (max-width: 768px)': {
      flexDirection: 'column',
      alignItems: 'center',
    },
  },
  logoTextContainer: {
    display: 'flex',
    alignItems: 'center',
    margin: '8px',
    '@media (max-width: 768px)': {
      flexDirection: 'row',
      alignItems: 'center',
    },
  },
  logo: {
    width: '24px',
    height: '24px',
    marginRight: '8px',
  },
  divider: {
    height: '20px',
    margin: '0 8px',
  },
  textContainer: {
    fontSize: '12px',
    color: '#333',
    '@media (max-width: 768px)': {
      textAlign: 'center',
    },
  },
});

function TopMenuBar() {
  const classes = useStyles();

  return (
      <div className={classes.fullWidthContainer}>
        <div className={classes.commandBar}>
          <div className={classes.logoTextContainer}>
            <Image src="logo-windows.png" alt="Microsoft Logo" className={classes.logo} />
            <div className={classes.textContainer}>
              <Text size={300} weight="medium">
                FinOps toolkit
              </Text>
            </div>
            <Divider vertical className={classes.divider} />
          </div>
        </div>
      </div>
  );
}

export default TopMenuBar;
