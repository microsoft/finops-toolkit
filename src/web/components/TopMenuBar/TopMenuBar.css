.ftk-topmenubar {
    width: 100dvw;
    height: 40px;
    padding: 0;
    background-color: var(--colorNeutralBackground1);
}

.ftk-topmenubar .ftk-commandbar {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    width: 100%;
    background-color: var(--colorNeutralBackground1Hover);
    overflow-x: hidden;
}

.ftk-topmenubar .ftk-logo-section {
    display: flex;
    align-items: center;
    margin: 8px;
}

.ftk-topmenubar .ftk-logo {
    width: 24px;
    height: 24px;
    margin-right: 8px;
}

.ftk-topmenubar .ftk-divider {
    height: 20px;
    margin: 0 8px;
    background-color: var(--colorNeutralForeground3);
}

@media (max-width: 768px) {
    .ftk-topmenubar .ftk-commandbar {
        flex-direction: column;
        align-items: center;
    }

    .ftk-topmenubar .ftk-logo-section {
        flex-direction: row;
        align-items: center;
    }
}
