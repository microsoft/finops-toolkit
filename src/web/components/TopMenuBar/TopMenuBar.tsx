import { Text, Image, Divider } from "@fluentui/react-components";
import "./TopMenuBar.css";
const TopMenuBar = () => {
  return (
    <header className="ftk-topmenubar">
      <div className="ftk-commandbar">
        <div className="ftk-logo-section">
          <Image src="logo-windows.png" alt="Microsoft Logo" className="ftk-logo" />
          <Text weight="medium">FinOps Toolkit</Text>
          <Divider vertical className="ftk-divider" />
        </div>
      </div>
    </header>
  );
};

export default TopMenuBar;
