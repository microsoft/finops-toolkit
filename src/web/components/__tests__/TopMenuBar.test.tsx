import { render, screen } from '@testing-library/react';
import TopMenuBar from '../TopMenuBar/TopMenuBar';
import React from 'react';

describe('TopMenuBar', () => {
    it('should render the logo and title', () => {
        render(<TopMenuBar />);

        // Check if the logo is rendered
        const logo = screen.getByAltText('Microsoft Logo');
        expect(logo).toBeInTheDocument();

        // Check if the title text is rendered
        const title = screen.getByText('FinOps toolkit');
        expect(title).toBeInTheDocument();
    });

    it('should render a divider between the logo and title', () => {
        render(<TopMenuBar />);

        // Check that the divider is present
        const divider = screen.getByRole('separator');
        expect(divider).toBeInTheDocument();
    });

    it('renders with accessible elements', () => {
        render(<TopMenuBar />);

        // Check if the logo has appropriate alt text
        const logo = screen.getByAltText('Microsoft Logo');
        expect(logo).toHaveAttribute('alt', 'Microsoft Logo');

        // Verify that the title text exists and contains the expected text
        const title = screen.getByText('FinOps toolkit');
        expect(title).toBeInTheDocument();
    });

    it('matches the snapshot', () => {
        const { asFragment } = render(<TopMenuBar />);
        expect(asFragment()).toMatchSnapshot();
    });
});
