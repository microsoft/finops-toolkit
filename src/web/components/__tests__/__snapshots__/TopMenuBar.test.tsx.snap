// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopMenuBar matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="fullWidthContainer"
  >
    <div
      class="commandBar"
    >
      <div
        class="logoTextContainer"
      >
        <img
          alt="Microsoft Logo"
          class="fui-Image logo ___1erkwlp_1hna692 fj3muxo f1akhkt f1aperda f1lxtadh f1fabniw f1ewtqcl f14t3ns0"
          src="logo-windows.png"
        />
        <div
          class="textContainer"
        >
          <span
            class="fui-Text ___1bdjl64_182myip fk6fouc fkhj508 f1i3iumi fdj6btp fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
          >
            FinOps toolkit
          </span>
        </div>
        <div
          aria-orientation="vertical"
          class="fui-Divider divider ___1esnkk0_1l2hrfi f122n59 f1ewtqcl f22iagw f1vx9l62 fqerorx f10pi13n fk6fouc fy9rknc figsok6 fwrc4pm f17mccla fyl8oag f16vkdww fhsnbul f1gw3sf2 f1ly5f7u f1s3tz6t f1wl9k8s f13zj6fq fkfq4zb f1vccso1 f1geml7w f1r7kh1m fjml6kk f16j7guv fx01ahm fl8d8yv fj1a37q f16gbxbe f1yq6w5o f9sc749 fhkwbjy flthirb f13hvwk3 f19onpk6 f1kyqvp9 fesgyo fzynn9s fekrn8e"
          role="separator"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
